src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -884,6 +914,20 @@ private IConfidentialClientApplication BuildConfidentialClientApplication(Merged
             return builder.ExecuteAsync(tokenAcquisitionOptions != null ? tokenAcquisitionOptions.CancellationToken : CancellationToken.None);
         }
 
+        internal static IDictionary<string, string> MergeExtraQueryParameters(
+            MergedOptions mergedOptions,
+            TokenAcquisitionOptions tokenAcquisitionOptions)
+        {
+            var mergedDict = tokenAcquisitionOptions!.ExtraQueryParameters;
```

Comment:
This will override the tokenAcquisitionOptions!.ExtraQueryParameters dictionary.
We'd want to create a new dictionnary (and you could then return `Dictionary<string, string>`

```csharp
var mergedDict = new Dictionary<string, string(tokenAcquisitionOptions!.ExtraQueryParameters);
```

Reply:
fixed